hadoop-tools/hadoop-distcp/src/test/java/org/apache/hadoop/tools/contract/AbstractContractDistCpTest.java

```diff
@@ -612,6 +634,9 @@ public void testDirectWrite() throws Exception {
 
   @Test
   public void testNonDirectWrite() throws Exception {
+    if (directWriteAlways()) {
+      skip("not needed");
```

Comment:
Nit: maybe move below `describe()`, or mention in skip message what is being skipped.

Reply:
actually, it should be in the previous test. So moved up. thanks for drawing my attention to it.